homeassistant/components/huawei_smart_logger/config_flow.py

```diff
@@ -0,0 +1,115 @@
+"""Config flow for HuaweiSmartLogger3000 integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import aiohttp
+from huawei_smart_logger.huawei_smart_logger import HuaweiSmartLogger3000API
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+async def validate_input(
+    hass: core.HomeAssistant, user_input: dict[str, Any]
+) -> dict[str, Any]:
+    """During configuration setup we will log in to validate credentials and server info."""
+    _LOGGER.debug("In validate input")
+
+    try:
+        hsl = HuaweiSmartLogger3000API(
+            user_input[CONF_USERNAME], user_input[CONF_PASSWORD], user_input[CONF_HOST]
+        )
+        await hsl.fetch_data()
+
+    except aiohttp.ClientError as e:
+        _LOGGER.error("Connection timed out to %s", user_input[CONF_HOST])
```

Comment:
How do we know that it's a timeout? We're just catching the general client error. I'd just say "Failed to connect to".
```suggestion
        _LOGGER.error("Failed to connect to %s", user_input[CONF_HOST])
```